File: src/main/java/me/nylestroke/nylemod/screen/ModScreenHandlers.java
package me.nylestroke.nylemod.screen;

import me.nylestroke.nylemod.NylemodExample;
import net.fabricmc.fabric.api.screenhandler.v1.ScreenHandlerRegistry;
import net.minecraft.screen.ScreenHandlerType;
import net.minecraft.util.Identifier;

public class ModScreenHandlers {

    public static ScreenHandlerType<MythrilBlasterScreenHandler> MYTHRIL_BLASTER_SCREEN_HANDLER;

    public static void registerAllScreenHandlers() {
        MYTHRIL_BLASTER_SCREEN_HANDLER =
                ScreenHandlerRegistry.registerSimple(new Identifier(NylemodExample.MOD_ID, "mythril_blaster"),
                        MythrilBlasterScreenHandler::new);
    }

}

File: src/main/kotlin/me/nylestroke/nylemod/screen/ModScreenHandlers.kt
package me.nylestroke.nylemod.screen

import me.nylestroke.nylemod.NylemodExample
import net.fabricmc.fabric.api.screenhandler.v1.ScreenHandlerRegistry
import net.minecraft.entity.player.PlayerInventory
import net.minecraft.screen.ScreenHandlerType
import net.minecraft.util.Identifier

object ModScreenHandlers {
    var MYTHRIL_BLASTER_SCREEN_HANDLER: ScreenHandlerType<MythrilBlasterScreenHandler>? = null
    fun registerAllScreenHandlers() {
        MYTHRIL_BLASTER_SCREEN_HANDLER = ScreenHandlerRegistry.registerSimple<MythrilBlasterScreenHandler>(Identifier(NylemodExample.MOD_ID, "mythril_blaster"),
            ScreenHandlerRegistry.SimpleClientHandlerFactory<MythrilBlasterScreenHandler> { syncId: Int, playerInventory: PlayerInventory ->
                MythrilBlasterScreenHandler(
                    syncId,
                    playerInventory
                )
            })
    }
}